src/providers/ratelimit-datasource.provider.ts

```diff
@@ -12,23 +13,37 @@ export class RatelimitDatasourceProvider
   implements Provider<Store | MemcachedStore | undefined>
 {
   constructor(
+    @inject.getter(RateLimitSecurityBindings.METADATA)
+    private readonly getMetadata: Getter<RateLimitMetadata>,
     @inject(CoreBindings.APPLICATION_INSTANCE)
     private readonly application: RestApplication,
     @inject(RateLimitSecurityBindings.CONFIG, {optional: true})
     private readonly config?: RateLimitOptions,
   ) {}
 
-  async value(): Promise<Store | MemcachedStore | undefined> {
+  value(): Promise<Store | MemcachedStore | undefined> {
+    return this.action();
+  }
+
+  async action(): Promise<Store | MemcachedStore | undefined> {
+    const metadata: RateLimitMetadata = await this.getMetadata();
+
+    // First check if rate limit options available at method level
+    const operationMetadata = metadata ? metadata.options : {};
+
+    // Create options based on global config and method level config
+    const opts = Object.assign({}, this.config, operationMetadata);
+
     if (this.config?.type === 'MemcachedStore') {
       return new MemcachedStore({
         client: this.config?.client,
-        expiration: (this.config?.windowMs ?? 60 * 1000) / 1000,
+        expiration: (opts.windowMs ?? 60 * 1000) / 1000,
       });
     } else if (this.config?.type === 'MongoStore') {
       return new MongoStore({
         uri: this.config?.uri,
         collectionName: this.config?.collectionName,
-        expireTimeMs: (this.config?.windowMs ?? 60 * 1000) / 1000,
+        expireTimeMs: (opts.windowMs ?? 60 * 1000) / 1000,
```

Comment:
why instead of config options is used

Reply:
config has default values, options is used to override these values for specific endpoints